src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -660,8 +664,109 @@ private SpanQuery getSpanQuery(
             protoSpanQuery.getSpanTermQuery();
         return new SpanTermQuery(
             new Term(protoSpanTermQuery.getField(), protoSpanTermQuery.getTextValue()));
+      case SPANMULTITERMQUERYWRAPPER:
+        com.yelp.nrtsearch.server.grpc.SpanMultiTermQueryWrapper protoSpanMultiTermQueryWrapper =
+            protoSpanQuery.getSpanMultiTermQueryWrapper();
+        return getSpanMultiTermQueryWrapper(protoSpanMultiTermQueryWrapper, state);
       default:
         throw new IllegalArgumentException("Unsupported Span Query: " + protoSpanQuery);
     }
   }
+
+  private SpanMultiTermQueryWrapper getSpanMultiTermQueryWrapper(
+      com.yelp.nrtsearch.server.grpc.SpanMultiTermQueryWrapper protoSpanMultiTermQueryWrapper,
+      IndexState state) {
+
+    com.yelp.nrtsearch.server.grpc.SpanMultiTermQueryWrapper.WrappedQueryCase wrappedQueryCase =
+        protoSpanMultiTermQueryWrapper.getWrappedQueryCase();
+
+    switch (wrappedQueryCase) {
+      case WILDCARDQUERY:
+        WildcardQuery wildcardQuery =
+            new WildcardQuery(
+                new Term(
+                    protoSpanMultiTermQueryWrapper.getWildcardQuery().getTerm().getField(),
+                    protoSpanMultiTermQueryWrapper.getWildcardQuery().getTerm().getText()));
+        return new SpanMultiTermQueryWrapper<>((MultiTermQuery) wildcardQuery);
+      case FUZZYQUERY:
+        FuzzyQuery fuzzyQuery = getFuzzyQuery(protoSpanMultiTermQueryWrapper);
+        return new SpanMultiTermQueryWrapper<>((MultiTermQuery) fuzzyQuery);
+      case PREFIXQUERY:
+        Query prefixQuery = getPrefixQuery(protoSpanMultiTermQueryWrapper.getPrefixQuery(), state);
+        if (prefixQuery instanceof MultiTermQuery) {
+          return new SpanMultiTermQueryWrapper<>((MultiTermQuery) prefixQuery);
+        }
+      case REGEXPQUERY:
+        RegexpQuery regexpQuery = getRegexpQuery(protoSpanMultiTermQueryWrapper);
+        if (regexpQuery instanceof MultiTermQuery) {
```

Comment:
Do we need all this type checking/casting. The queries should all inherit from `MultiTermQuery`.

Reply:
I only needed to keep the casting for `prefixQuery` as it was `Query` unlike the other ones.